### ISSUE TITLE
Skeleton Loader on Task Details Page Stuck Forever for Missing or Failed API Response

### ISSUE THREAD

**Issue by AnujChhikara** [Real-Dev-Squad/website-dashboard#925]
### Issue Description

When navigating to the Task Details page, the skeleton loader is displayed while data is being fetched from the API. However, if the API returns an error (e.g., task not found or a server issue), the skeleton loader remains visible indefinitely. The user is not informed about the error, leading to confusion.

### Expected Behavior

If the API returns an error, the Task Details page should display an appropriate error message, such as:
- "Task not found" (for a 404 error).
- "Error fetching task details. Please try again later." (for other errors).
The skeleton loader should stop displaying once an error occurs.
### Current Behavior

When the API fails to fetch the task details:

- The skeleton loader remains visible indefinitely.
- No error message is displayed to the user.
### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
-  Go to this [task details page](https://dashboard.realdevsquad.com/task-requests/details/?id=voImPmPhqrL8Wa6x) 
-  I have put wrong task id in the params and observe that the skeleton loader remains visible indefinitely without showing an error message.

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [x] Medium
 - [ ] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.
### Additional Information

 - Backend Issue related to this [link](https://github.com/Real-Dev-Squad/website-backend/issues/2290)

**Comment by abbi4code** [comment]
Can i work on this issue ? Can you please assign the issue to me ? @Achintya-Chatterjee

**Comment by AnujChhikara** [comment]
> Can i work on this issue ? Can you please assign the issue to me ? @Achintya-Chatterjee

Hello @abbi4code can you please write this in the discord server

**Comment by Achintya-Chatterjee** [comment]
@abbi4code are you in the RDS discord server?

**Comment by abbi4code** [comment]
@Achintya-Chatterjee No, Im not in the RDS Discord server, but I would love to join.

**Comment by abbi4code** [comment]
> > Can i work on this issue ? Can you please assign the issue to me ? @Achintya-Chatterjee
> 
> Hello @abbi4code can you please write this in the discord server

how can i join the discord server ?

**Comment by Achintya-Chatterjee** [comment]
> > > Can i work on this issue ? Can you please assign the issue to me ? @Achintya-Chatterjee
> > 
> > 
> > Hello @abbi4code can you please write this in the discord server
> 
> how can i join the discord server ?

@abbi4code for this you have to contact @ankushdharkar
